android/app/src/main/java/it/pagopa/io/app/MainApplication.java

```diff
@@ -3,103 +3,69 @@
 import com.facebook.react.ReactApplication;
 import com.facebook.react.ReactNativeHost;
 import com.facebook.react.ReactPackage;
-import com.facebook.react.config.ReactFeatureFlags;
+import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint;
+import com.facebook.react.defaults.DefaultReactNativeHost;
 import com.facebook.soloader.SoLoader;
-import it.pagopa.io.app.newarchitecture.MainApplicationReactNativeHost;
 import it.ipzs.cieidsdk.native_bridge.CiePackage;
 import com.reactnativecommunity.art.ARTPackage;
 import com.facebook.react.bridge.JSIModulePackage;
-import com.swmansion.reanimated.ReanimatedJSIModulePackage;
 
 import com.facebook.react.PackageList;
 
 import android.app.Application;
 import android.content.Context;
 import com.facebook.react.ReactInstanceManager;
-import java.lang.reflect.InvocationTargetException;
 
 import java.util.List;
 
 public class MainApplication extends Application implements ReactApplication {
 
 
   private final ReactNativeHost mReactNativeHost =
-      new ReactNativeHost(this) {
+      new DefaultReactNativeHost(this) {
         @Override
         public boolean getUseDeveloperSupport() {
           return BuildConfig.DEBUG;
         }
 
-    @Override
-    protected String getJSMainModuleName() {
-      return "index";
-    }
-
-    @Override
-    protected JSIModulePackage getJSIModulePackage() {
-      return new ReanimatedJSIModulePackage();
-    }
+        @Override
+        protected String getJSMainModuleName() {
+          return "index";
+        }
 
-    @Override
-    protected List<ReactPackage> getPackages() {
-      List<ReactPackage> packages = new PackageList(this).getPackages();
-      packages.add(new CiePackage());
-      packages.add(new ARTPackage());
-      return packages;
-    }
-  };
+        @Override
+        protected List<ReactPackage> getPackages() {
+          List<ReactPackage> packages = new PackageList(this).getPackages();
+          packages.add(new CiePackage());
+          packages.add(new ARTPackage());
+          return packages;
+        }
 
-  private final ReactNativeHost mNewArchitectureNativeHost =
-      new MainApplicationReactNativeHost(this);
+        @Override
+        protected boolean isNewArchEnabled() {
+          return BuildConfig.IS_NEW_ARCHITECTURE_ENABLED;
```

Comment:
Actually, we don't have `IS_NEW_ARCHITECTURE_ENABLED` in our build.gradle, Am I right? What about adding it and explicitly set it to `false`. 

What about restoring this?

```groovy
buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
```

What do you think?

Reply:
the attribute is obtained directly from the params in the `gradle.properties` such as the hermes one.

you can find info about enabling the new arrch [here](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-apps.md#for-existing-apps-1)

Reply:
the code you suggested is already defined inside the `@react-native/gradle-plugin`, that's why we don't need to specify by our own anymore